Datastore/tests/System/FilterTest.php

```diff
@@ -0,0 +1,221 @@
+        $this->assertEquals($results[0]['Name'], 'Hersch');
+        $this->assertEquals($results[1]['Name'], 'Davy');
+        $transaction->commit();
+    }
+
+    /**
+     * @dataProvider defaultDbClientProvider
+     */
+    public function testMixOfOrAndViaTransaction(DatastoreClient $client)
+    {
+        $this->skipEmulatorTests();
+        $filter = Filter::and([
+            Filter::where('Age', '<', 26),
+            Filter::or([
+                Filter::and([
+                    Filter::where('Age', '>', 23),
+                    Filter::where('Age', '<', 30)
+                ]),
+                Filter::and([
+                    Filter::where('Age', '>', 25),
+                    Filter::where('Age', '<', 31)
+                ]),
+            ])
+        ]);
+        $query = $client->query()
+            ->kind(self::$kind)
+            ->filter($filter);
+
+        $transaction = $client->transaction();
+        $results = $this->runQueryAndSortResults($transaction, $query);
+        $this->assertEquals(count($results), 1);
+        $this->assertEquals($results[0]['Name'], 'Eldredge');
+        $transaction->commit();
+    }
+
+    private static function getInitialData()
+    {
+        return [[
+            "Name" => "Hersch",
+            "Age" => 21,
+            "FavouriteColor" => "Purple",
+            "Gender" => "Male"
+        ], [
+            "Name" => "Davy",
+            "Age" => 23,
+            "FavouriteColor" => "Turquoise",
+            "Gender" => "Male"
+        ], [
+            "Name" => "Eldredge",
+            "Age" => 25,
+            "FavouriteColor" => "Violet",
+            "Gender" => "Male"
+        ], [
+            "Name" => "Rickey",
+            "Age" => 26,
+            "FavouriteColor" => "Crimson",
+            "Gender" => "Bigender"
+        ], [
+            "Name" => "Arron",
+            "Age" => 30,
+            "FavouriteColor" => "Red",
+            "Gender" => "Male"
+        ], [
+            "Name" => "Hi",
+            "Age" => 31,
+            "FavouriteColor" => "Teal",
+            "Gender" => "Male"
+        ], [
+            "Name" => "Chandler",
+            "Age" => 33,
+            "FavouriteColor" => "Turquoise",
+            "Gender" => "Male"
+        ], [
+            "Name" => "Juliette",
+            "Age" => 36,
+            "FavouriteColor" => "Purple",
+            "Gender" => "Female"
+        ], [
+            "Name" => "Mufi",
+            "Age" => 36,
+            "FavouriteColor" => "Fuscia",
+            "Gender" => "Female"
+        ], [
+            "Name" => "Karrah",
+            "Age" => 39,
+            "FavouriteColor" => "Violet",
+            "Gender" => "Female"
+        ]];
+    }
+
+    private function runQueryAndSortResults($client, $query)
+    {
+        $results = iterator_to_array($client->runQuery($query));
+        usort($results, function ($a, $b) {
+            return $a['Name'] < $b['Name'];
+        });
```

Comment:
You should not need to sort on the results, since it can be done by the backend / emulator.
```suggestion
        $query->order('Name');
        $results = iterator_to_array($client->runQuery($query));
```

Reply:
Yes that indeed is a way. I was also considered doing this but then noticed that we use this helper method way in RunQueryTest. So went ahead with this way to be consistent within the library.

Should we do the `order`ing way?

Reply:
Okay, lets skip it. Not a dealbreaker.